naga/src/back/spv/block.rs

```diff
@@ -362,21 +362,47 @@ impl<'w> BlockContext<'w> {
                             base_id,
                             base_ty,
                             index_id,
-                            ty_element,
+                            LookupType::Handle(ty_element),
+                            block,
+                        )?;
+                        self.function.internal_variables.push(variable);
+                        id
+                    }
+                    crate::TypeInner::Matrix { scalar, rows, .. } => {
+                        let index_id = self.cached[index];
+                        let base_id = self.cached[base];
+                        let base_ty = match self.fun_info[base].ty {
+                            TypeResolution::Handle(handle) => handle,
+                            TypeResolution::Value(_) => {
+                                return Err(Error::Validation(
+                                    "Array types should always be in the arena",
```

Comment:
```suggestion
                                    "Matrix types should always be in the arena",
```

I don't think this is always true:

https://github.com/gfx-rs/wgpu/blob/43cb730d588a06ae64d91ce98afbef09d5613596/naga/src/proc/typifier.rs#L98-L98

Reply:
This was copy&paste from array, I will check and fix.